### PR TITLE
Add download/upload feature for domain pattern block list

### LINKED ISSUE #11775: Support import/export for domain blocks
Offshoot of https://github.com/friendica/friendica/issues/11773

This would enable anyone downloading the domain block list available at `/friendica` under a structured file format, and administrators to import such a file to either extend or replace their current block list.

### PR DESCRIPTION
**Pull request by MrPetovan**
Closes #11775

### CODE REVIEW

**Review by Quix0r — APPROVED** [review]
Looks fine to me.

**Comment by tobiasd** [comment]
I would prefer to have the two exported lists of blocked servers in the same format. The console command is using CSV, here you use JSON if I see correctly. Personally I prefer CSV as it is more human readable, but as this is mostly meant as system-interchange format and not human readable, I would not mind if the console tool was switched to JSON as well.

On the other hand, contact list exports are done as CSV file (as do other platforms in the Fediverse) so maybe stick with CSV as export format in all cases.

**Comment by MrPetovan** [comment]
I wasn't aware there was a console command to export the blocklist, I'll switch the UI pages to CSV.

File: src/Module/Admin/Blocklist/Server/Import.php [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\Admin\Blocklist\Server;
+
+use Friendica\Core\Renderer;
+use Friendica\DI;
```
Comment (nupplaphil) [inline src/Module/Admin/Blocklist/Server/Import.php]: Please replace all `DI::*` with a proper constructor initialization:)

**Review by nupplaphil — CHANGES_REQUESTED** [review]

File: src/Module/Admin/Blocklist/Server/Import.php (hunk 1 above)
Comment (MrPetovan) [reply src/Module/Admin/Blocklist/Server/Import.php]: I forgot it works with Modules now!

**Review by nupplaphil — APPROVED** [review]

**Pull request merged**